includes/admin/admin-actions.php

```diff
@@ -568,13 +568,98 @@ function give_donation_import_callback() {
 
 	$percentage              = ( 100 / ( $total_ajax + 1 ) ) * $current;
 	$json_data['percentage'] = $percentage;
-
+	
 	$json_data = apply_filters( 'give_import_ajax_responces', $json_data, $fields );
 	wp_die( json_encode( $json_data ) );
 }
 
 add_action( 'wp_ajax_give_donation_import', 'give_donation_import_callback' );
 
+/**
+ * Load core settings import ajax callback
+ * Fire when importing from JSON start
+ *
+ * @since  1.8.16
+ *
+ * @return json $json_data
+ */
+
+function give_core_settings_import_callback() {
+	$fields = isset( $_POST['fields'] ) ? $_POST['fields'] : null;
+	parse_str( $fields, $fields );
+
+	$json_data['success'] = false;
+
+	/**
+	 * Filter to Modify fields that are being pass by the ajax before importing of the core setting start.
+	 *
+	 * @access public
+	 *
+	 * @since 1.8.16
+	 *
+	 * @param array $fields
+	 *
+	 * @return array $fields
+	 */
+	$fields = (array) apply_filters( 'give_import_core_settings_fields', $fields );
+
+	$file_name = ( ! empty( $fields['file_name'] ) ? give_clean( $fields['file_name'] ) : false );
+
+	if ( ! empty( $file_name ) ) {
+		$type = ( ! empty( $fields['type'] ) ? give_clean( $fields['type'] ) : 'merge' );
+
+		// Get the json data from the file and then alter it in array format
+		$json_string   = give_get_core_settings_json( $file_name );
+		$json_to_array = json_decode( $json_string, true );
+
+		// get the current settign from the options table.
+		$host_give_options = get_option( 'give_settings', array() );
+
+		// Save old settins for backup.
+		update_option( 'give_settings_old', $host_give_options );
+
+		/**
+		 * Filter to Modify Core Settings that are being going to get import in options table as give settings.
+		 *
+		 * @access public
+		 *
+		 * @since 1.8.16
+		 *
+		 * @param array $json_to_array Setting that are being going to get imported
+		 * @param array $type Type of Import
+		 * @param array $host_give_options Setting old setting that used to be in the options table.
+		 * @param array $fields Data that is being send from the ajax
+		 *
+		 * @return array $json_to_array Setting that are being going to get imported
+		 */
+		$json_to_array = (array) apply_filters( 'give_import_core_settings_data', $json_to_array, $type, $host_give_options, $fields );
+
+		update_option( 'give_settings', $json_to_array );
+
+		$json_data['success'] = true;
+	}
+
+	$json_data['percentage'] = 100;
+
+	/**
+	 * Filter to Modify core import setting page url
+	 *
+	 * @access public
+	 *
+	 * @since 1.8.16
+	 *
+	 * @return array $url
+	 */
+	$json_data['url'] = give_import_page_url( (array) apply_filters( 'give_import_core_settings_success_url', array(
+		'step'          => ( empty( $json_data['success'] ) ? '1' : '3' ),
+		'importer-type' => 'import_core_setting',
+		'success'       => ( empty( $json_data['success'] ) ? '0' : '1' ),
+	) ) );
+
+	wp_die( json_encode( $json_data ) );
```

Comment:
@raftaar1191 Why did not you use `wp_send_json` fx

Reply:
I don't know y I did that. I guess I just copy the code and past it right over here 

Fixed now @ravinderk for pointing this out to me